ethcore/engines/authority-round/src/lib.rs

```diff
@@ -1095,10 +1092,7 @@ impl AuthorityRound {
 			None => return Ok(Vec::new()), // We are not a validator, so we shouldn't call the contracts.
 		};
 		let our_addr = signer.address();
-		let client = self.client.read().as_ref().and_then(|weak| weak.upgrade()).ok_or_else(|| {
-			debug!(target: "engine", "Unable to prepare block: missing client ref.");
-			EngineError::RequiresClient
-		})?;
+		let client = self.client_arc("Unable to prepare block")?;
 		let full_client = client.as_full_client()
```

Comment:
Not your code but I think `as_full_client()` is a misleading name; maybe `upcast_client()` would be better.

Reply:
That would touch eight files; maybe it would be better as a separate PR, to keep this one focused on Authority Round?